sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1618,10 +1618,13 @@ ProgramManager::compile(const device_image_plain &DeviceImage,
   // TODO: Set spec constatns here.
 
   // TODO: Handle zero sized Device list.
+
+  const RTDeviceBinaryImage *ImgPtr = InputImpl->get_bin_image_ref();
```

Comment:
maybe just a matter of style, but mightn't `const char *compileOptions = InputImpl->get_bin_image_ref()->getCompileOptions();` be slightly better than this juggling of `*ImgPtr` and `&Img` ?    The `Img` var is only used one place (though there is another one of these jugglings down in the loop in 1660

Reply:
Yes, I agree with you. I'll change it.